packages/core/src/StateNode.ts

```diff
@@ -50,108 +29,43 @@ import {
   ActionObject,
   Mapper,
   PropertyMapper,
-  SendAction,
   NullEvent,
-  Guard,
-  GuardPredicate,
-  GuardMeta,
-  MachineConfig,
-  PureAction,
-  DoneEventObject,
-  SingleOrArray,
-  LogAction,
-  SendActionObject,
-  SpecialTargets,
-  RaiseAction,
   SCXML,
-  RaiseActionObject,
-  Typestate,
-  TransitionDefinitionMap,
-  DelayExpr,
-  ActivityActionObject,
-  ActivityDefinition
+  TransitionDefinitionMap
 } from './types';
 import { matchesState } from './utils';
-import { State, stateValuesEqual } from './State';
+import { State } from './State';
 import * as actionTypes from './actionTypes';
+import { toActionObject } from './actions';
+import { isLeafNode } from './stateUtils';
 import {
-  send,
-  cancel,
-  after,
-  raise,
-  done,
-  doneInvoke,
-  error,
-  toActionObject,
-  resolveSend,
-  initEvent,
-  toActionObjects,
-  resolveLog,
-  resolveRaise,
-  start,
-  stop,
-  toActivityDefinition
-} from './actions';
-import { IS_PRODUCTION } from './environment';
-import { DEFAULT_GUARD_TYPE, STATE_DELIMITER } from './constants';
-import {
-  getValue,
-  getConfiguration,
-  has,
-  getChildren,
-  getAllStateNodes,
-  isInFinalState,
-  isLeafNode
-} from './stateUtils';
-import { createInvocableActor } from './Actor';
-
-const NULL_EVENT = '';
-const STATE_IDENTIFIER = '#';
-const WILDCARD = '*';
+  getDelayedTransitions,
+  formatTransitions,
+  getCandidates,
+  getStateNodeById,
+  getRelativeStateNodes,
+  nodesFromChild,
+  evaluateGuard,
+  isStateId
+} from './nodeUtils';
+import { MachineNode } from './MachineNode';
+import { STATE_DELIMITER } from './constants';
+
+export const NULL_EVENT = '';
```

Comment:
those constants are now defined both in StateNode and in MachineNode, they should be just moved to `./constants.ts`